File: folkmq-lang-java/folkmq-test/src/test/java/labs/HaTest.java
package labs;

import org.noear.folkmq.FolkMQ;
import org.noear.folkmq.client.MqClient;
import org.noear.folkmq.client.MqMessage;

public class HaTest {
    public static void main(String[] args) throws Exception {
        //客户端
        MqClient client = FolkMQ.createClient("folkmq:ws://127.0.0.1:18603?ak=ak1&sk=sk1",
                        "folkmq:ws://127.0.0.1:18703?ak=ak1&sk=sk1")
                .nameAs("demoapp")
                .connect();

        client.subscribe("demo", (msg) -> {
            System.out.println(msg.getBodyAsString());
        });

        long i = 0;
        while (true) {
            client.publishAsync("demo", new MqMessage("test-" + (i++)));
        }
    }
}

File: folkmq-lang-js/folkmq-test/demo02-node-ws/_ha_test.js
const {FolkMQ} = require("@noear/folkmq");

async function main() {
    const client =  await FolkMQ.createClient("folkmq:ws://127.0.0.1:18603?ak=ak1&sk=sk1",
        "folkmq:ws://127.0.0.1:18703?ak=ak1&sk=sk1")
        .nameAs("demoapp")
        .connect();

    //订阅
    client.subscribe("demo", null, true, message => {
        console.log(message.getBodyAsString());
    });

    //发布
    let i=0;
    while (true) {
        i++
        client.publish("demo", FolkMQ.newMessage("hot-" + i));
    }
}

main();